src/serde/deserialize_program.rs

```diff
@@ -137,6 +163,49 @@ impl<'de> de::Visitor<'de> for ReferenceIdsVisitor {
     }
 }
 
+struct ValueAddressVisitor;
+
+impl<'de> de::Visitor<'de> for ValueAddressVisitor {
+    type Value = ValueAddress;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("Could not deserialize hexadecimal string")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        // regex for capturing the register the address is referenced to
+        let register_re = Regex::new(r".p").unwrap();
```

Comment:
It's better to restrict the matched character to the correct registers:
```suggestion
        let register_re = Regex::new(r"[af]p").unwrap();
```
The more specific the regex, the more efficient the resulting state machine and its compilation.

Reply:
(Typically)

Reply:
done!